### ISSUE TITLE
Incorrect error about missing annotation of destructured parameter

### ISSUE THREAD

**Issue by nwaywood** [gajus/eslint-plugin-flowtype#211]
I have this function in my code

```
const renderItem = ({ title }) => <li>{title}</li>;
```

and the plugin is giving me this lint error

![image](https://cloud.githubusercontent.com/assets/3773641/24433475/b5a95548-1474-11e7-98c3-5cc8289ef28d.png)

but flow does not yet support type annotations within destructured parameters. See [this](https://github.com/facebook/flow/issues/235) issue for more information.

I think this error should only be shown for non-destructured parameters

**Comment by danharper** [comment]
You can still type the object:

```js
const renderItem = ({ title }: { title: string }) => <li>{title}</li>;
```
